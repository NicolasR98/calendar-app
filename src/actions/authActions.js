import { types } from "../types/types";
import { fetchWithoutToken } from "../helpers/fetch";
import Swal from "sweetalert2";

export const startLogin = (email, password) => {
    const auxParams = {
        endpoint: 'auth',
        data: { email, password },
        method: 'POST'
    };

    return async (dispatch) => {
        const resp = await fetchWithoutToken(auxParams);
        const body = await resp.json();

        if (body.ok) {
            const user = {
                uid: body.uid,
                name: body.name,
            };

            localStorage.setItem('token', body.token);
            localStorage.setItem('token-init-date', new Date().getTime());

            dispatch(login(user));
        } else {
            Swal.fire('Error', body.msg, 'error');
        }
    };
};

const login = (user) => ({
    type: types.authLogin,
    payload: user
});