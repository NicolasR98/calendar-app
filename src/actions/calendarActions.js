import { types } from "../types/types";
import { fetchWithToken } from "../helpers/fetch";

export const eventStartAddNew = (event) => {
    return async (dispatch, getState) => {
        const auxParams = {
            endpoint: 'events',
            data: event,
            method: 'POST'
        };
        const { uid, name } = getState().auth;

        try {
            const resp = await fetchWithToken(auxParams);
            const body = await resp.json();

            if (body.ok) {
                event.id = body.event.id;
                event.user = {
                    _id: uid,
                    name,
                };
                dispatch(eventAddNew(event));
            };

        } catch (error) {
            console.error(error);
        }

    };
};

const eventAddNew = (event) => ({
    type: types.calendarAddNew,
    payload: event,
});

export const eventSetActive = (event) => ({
    type: types.calendarSetActive,
    payload: event,
});

export const eventUpdate = (event) => ({
    type: types.calendarUpdate,
    payload: event,
});

export const eventDelete = () => ({ type: types.calendarDelete });

export const eventClearActive = () => ({ type: types.calendarClearActive });