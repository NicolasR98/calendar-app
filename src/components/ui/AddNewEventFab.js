import React from 'react';
import { useDispatch } from 'react-redux';
import { openModal } from '../../actions/uiActions';

export const AddNewEventFab = () => {
    const dispatch = useDispatch();

    const handleOpenModal = () => {
        dispatch(openModal());
    };

    return (
        <button
            className='btn btn-primary fab'
            onClick={handleOpenModal}
        >
            <i className='fas fa-plus' />
        </button>
    );
};
