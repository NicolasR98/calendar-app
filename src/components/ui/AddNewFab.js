import React from 'react';
import { useDispatch } from 'react-redux';
import { openModal } from '../../actions/uiActions';

export const AddNewFab = () => {
    const dispatch = useDispatch();

    const handleOpenModal = () => {
        dispatch(openModal());
    };

    return (
        <button
            className='btn btn-primary fab'
            onClick={handleOpenModal}
        >
            <i className='fas fa-plus' />
        </button>
    );
};
