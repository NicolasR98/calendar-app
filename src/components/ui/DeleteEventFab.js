import React from 'react';
import { useDispatch } from 'react-redux';
import { eventStartDelete } from '../../actions/calendarActions';

export const DeleteEventFab = () => {
    const dispatch = useDispatch();

    const handleDeleteNote = () => {
        dispatch(eventStartDelete());
    };

    return (
        <button
            onClick={handleDeleteNote}
            className='btn btn-danger fab fab--delete'
        >
            <i className='fas fa-trash' />
        </button>
    );
};
