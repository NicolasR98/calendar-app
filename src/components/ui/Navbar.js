import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/authActions';
import { eventClearAll } from '../../actions/calendarActions';

export const Navbar = () => {
    const dispatch = useDispatch();
    const { name } = useSelector(state => state.auth);

    const handleLogout = () => {
        dispatch(startLogout());
        dispatch(eventClearAll());
    };

    return (
        <div className='navbar navbar-dark bg-dark mb-4 px-3 sticky-top'>
            <span className='navbar-brand'>
                {name}
            </span>

            <button
                className='btn btn-outline-danger'
                onClick={handleLogout}
            >
                <i className='fas fa-sign-out-alt' />
                <span>Logout</span>
            </button>
        </div>
    );
};
