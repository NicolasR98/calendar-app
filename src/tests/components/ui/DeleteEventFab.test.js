import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/calendarActions';

jest.mock('../../../actions/calendarActions', () => ({
    eventStartDelete: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <DeleteEventFab />
    </Provider>
);

describe('Tests on <DeleteEventFab />', () => {
    test('Should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should invoke eventStartDelete when onClick', () => {
        wrapper.find('.fab--delete').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalledTimes(1);
    });
});