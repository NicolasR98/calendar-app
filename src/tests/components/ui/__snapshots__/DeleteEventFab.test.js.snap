// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <DeleteEventFab /> Should display correctly 1`] = `
<button
  className="btn btn-danger fab fab--delete"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
</button>
`;
